test/lib/ansible_test/_internal/commands/sanity/validate_modules.py

```diff
@@ -159,7 +159,11 @@ def test(self, args: SanityConfig, targets: SanityTargets, python: PythonConfig)
                 temp_dir = process_scoped_temporary_directory(args)
 
                 with tarfile.open(path) as file:
-                    file.extractall(temp_dir)
+                    # deprecated: description='extractall fallback without filter' python_version='3.11'
+                    if hasattr(tarfile, 'tar_filter'):
+                        file.extractall(temp_dir, filter='tar')  # type: ignore[call-arg]
```

Comment:
The data filter should be usable here, since the symlinks we're using don't reference parent directories.